src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1247 @@
+}
+
+/**
+ * If an instruction fetch got detected on a monitored page, this callback is executed
+ *
+ */
+static event_response_t execute_faulted_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    PRINT_DEBUG("[HYPERBEE] Caught X on PA 0x%lx, frame VA %llx, CR3 %lx\n", info->trap_pa,
+                (unsigned long long) info->regs->rip, info->regs->cr3);
+
+    //load the trap data
+    auto* ef_data = (struct fault_data_struct*) info->trap->data;
+
+    //Verify the program leading to the execution of this trap is the one we are filtering for (if we do).
+    //The filtering could limit (and therefore focus) the monitoring and gives a speedup in such possibly uninteresting cases.
+    // If it does not match the filter, delete this trap (and don't replace it)
+    if (ef_data->plugin->hyperbee_filter_executable[0])
+    {
+        if (strcasestr(info->proc_data.name, ef_data->plugin->hyperbee_filter_executable) == nullptr)
+        {
+            //Removes this trap and frees the memory
+            ef_data->plugin->traps.erase(info->trap);
+            drakvuf_remove_trap(drakvuf, info->trap, remove_trap_cb);
+            PRINT_DEBUG("[HYPERBEE] Removed outdated trap for PA 0x%lx", info->trap_pa);
+            return VMI_EVENT_RESPONSE_NONE;
+        }
+    }
+
+    //Initialize a lot of data
+    //The start of the VAD-Node. This is not necessarily the starting va of the page that is analysed.
+    addr_t vad_node_base = 0;
+    //The end address of the VAD-Node. The VAD-Node is not limited to a PAGE_SIZE.
+    addr_t vad_node_end = 0;
+    //The name of the dll
+    unicode_string_t* dll_name = nullptr;
+    //default name for the VAD / Dll
+    char missing_data[] = "(null)";
+    //default name, when the vad node represents no memory mapped file.
+    char alloc_memory[] = "(no-mapped-file)";
+    //The name of the application
+    char* dll_name_str = missing_data;
+    //The checksum of the current analyzed memory (vad or page)
+    const gchar* chk_str = nullptr;
+
+    //The size of current analyzed memory part (the amount of bytes written to disk)
+    size_t memory_size = 0;
+    //The whole path of the dumpfile
+    char* file_path_memory_dump = nullptr;
+    //The path to the file containing the metadata
+    char* file_path_meta_data = nullptr;
+    //The filename of which the memory dump and meta data file paths are built. it is a concatenation of the start address and the hash.
+    char* file_name_stem = nullptr;
+    //Is used to find a checksum in the set of already analyzed memory parts
+    std::unordered_map<std::string, std::string>::iterator memory_hash_identifier;
+
+    //initial value for malware. This can be set by an optional integrated classifier. If this is set to true (or during the execution of this method), the page dump will be stored always.
+    //TODO FutureWork: if the classifier is integrated, set it to false by default, and let this be set by the classifier
+    bool malware = MALWARE_DEFAULT;
```

Comment:
This looks like it neeeds a struct of it's own (or two).

Reply:
Done!